### PR TITLE
Add build script

### PR DESCRIPTION
**Pull request by eiglow**
I use this script myself often when hacking around with Lightspark, and it fits my needs pretty well. Maybe it needs some more options before being added? What do you think? Assuming you think this is a good idea at all.

In this PR I also updated the readme to make use of this script. Later I might try to make a `prepare` script which automatically grabs dependencies.

### CODE REVIEW

**Comment by masatake** [comment]
How about adding --debug, -h and --help options?
In my experience, -h and --help options are always helpful if a script takes arugments and/or options.

**Comment by eiglow** [comment]
Do you think it would be a good idea to add more build types, like LeanDebug, RelWithDebInfo, and Profile? I was also thinking about adding an option to specify how many threads to use, overriding the automatic detection.

**Comment by masatake** [comment]
I'm not sure whether @dbluelle likes this or not. (I'm just watching this important project.)
However, if you really want to push this for merging, I think updating README.md and .travis.yml to adapt to your script are needed. Just a comment.

**Comment by dbluelle** [comment]
2 comments:
- I don't know if cmake properly handles building debug and release versions from the same build dir, so it might be better to modify the scrpt to use different build dirs.
- I think we should keep the "normal" build instructions in the readme and just add the "build" script as an additional method. (I don't like it when projects have no real build instructions and you have to get every step out of a build script, but that's just me...)

File: build [hunk 1]
```diff
@@ -3,29 +3,40 @@
...
 
 case "$1" in
-	-h | --help)
-		print_help
-		exit 0
-		;;
 	-d | --debug)
 		mkdir -p obj-debug
 		cd obj-debug
+		check_cmake
```
Comment (masatake) [inline build]: How about introducing check_cmds() ?

check_cmds() checks the availability of the commands needed for building the target like cmake and make.
Running check_cmds before making a directory may be better. If one doesn't install cmake, making a directory is a waste of CPU time. If one doesn't install make, running cmake is also a waste of CPU time.

Comment (eiglow) [reply build]: Good idea.

**Pull request merged**